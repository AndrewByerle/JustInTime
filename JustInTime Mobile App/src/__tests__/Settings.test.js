import { render, screen } from '@testing-library/react-native';
import SettingsScreen from './../../src/navigation/screens/Settings';
import renderer from 'react-test-renderer';

jest.useFakeTimers();

test('integration check that title shows on settings page', () => {
    render(<SettingsScreen />);
    expect(screen.getByText('Settings')).toBeTruthy();
});

test('check that date bar shows on settings page', () => {
    render(<SettingsScreen />);
    expect(
        screen.getByText(`${new Date().getDate()}`, {
            exact: false,
        })
    ).toBeTruthy();
});

test('check notifications settings render properly', () => {
    render(<SettingsScreen />);
    expect(screen.getByText('Allow notifications')).toBeTruthy();
    expect(screen.getByText('Subscribed notifications:')).toBeTruthy();
});

test('check feed settings render properly', () => {
    render(<SettingsScreen />);
    expect(screen.getByText('Show in feed:')).toBeTruthy();
});

test('check for same settings page snapshot', () => {
    const tree = renderer.create(<SettingsScreen />).toJSON();
    expect(tree).toMatchSnapshot();
});
