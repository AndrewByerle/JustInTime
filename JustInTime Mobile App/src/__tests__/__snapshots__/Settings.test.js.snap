// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check for same updates page snapshot 1`] = `
<View>
  <View
    style={
      [
        {
          "paddingHorizontal": 24,
        },
        {
          "paddingTop": 56,
        },
        {
          "paddingBottom": 8,
        },
      ]
    }
  >
    <Text
      style={
        {
          "color": "#4B9CD3",
          "fontSize": 24,
        }
      }
    >
      Monday, November 28
    </Text>
  </View>
  <View
    style={
      {
        "paddingHorizontal": 24,
        "paddingVertical": 0,
      }
    }
  >
    <Text
      style={
        [
          {
            "fontSize": 36,
          },
          {
            "fontWeight": "bold",
          },
          {
            "color": "#4B9CD3",
          },
        ]
      }
    >
      Settings
    </Text>
  </View>
</View>
`;
